packages/e2e/config/wdio.browser.conf.ts

```diff
@@ -0,0 +1,62 @@
+import { config as baseConfig } from './wdio.conf.ts';
+import path from 'path';
+import url from 'url';
+import fs from 'fs/promises';
+import { getChromeExtensionPath, getFirefoxExtensionPath } from '../utils/extension-path.ts';
+
+const __dirname = url.fileURLToPath(new URL('.', import.meta.url));
+const archiveName = process.env.__FIREFOX__ ? 'extension.xpi' : 'extension.zip';
+const extPath = path.join(__dirname, `../../../dist-zip/${archiveName}`);
+const bundledExtension = (await fs.readFile(extPath)).toString('base64');
+const isCI = process.env.CI === 'true';
+
+const chromeCapabilities = {
+  browserName: 'chrome',
+  browserVersion: 'latest',
+  acceptInsecureCerts: true,
+  'goog:chromeOptions': {
+    args: [
+      '--disable-web-security',
+      '--disable-gpu',
+      '--no-sandbox',
+      '--disable-dev-shm-usage',
+      ...(isCI ? ['--headless'] : []),
+    ],
+    prefs: { 'extensions.ui.developer_mode': true },
+    extensions: [bundledExtension],
+  },
+};
+
+const firefoxCapabilities = {
+  browserName: 'firefox',
+  browserVersion: 'latest',
+  acceptInsecureCerts: true,
+  'moz:firefoxOptions': {
+    args: [...(isCI ? ['--headless'] : [])],
+  },
+};
+
+export const config: WebdriverIO.Config = {
+  ...baseConfig,
+  capabilities: process.env.__FIREFOX__ === 'true' ? [firefoxCapabilities] : [chromeCapabilities],
+  maxInstances: isCI ? 10 : 1,
+  execArgv: isCI ? [] : ['--inspect'],
+  before: async (capabilities: WebdriverIO.Capabilities, _specs, browser: WebdriverIO.Browser) => {
```

Comment:
You're using only browserName, you can do it:

```suggestion
  before: async ({browserName}: WebdriverIO.Capabilities, _specs, browser: WebdriverIO.Browser) => {
```

This should simplify